public/app/features/alerting/unified/utils/settings.ts

```diff
@@ -2,7 +2,11 @@ import { AlertmanagerChoice, GrafanaAlertingConfiguration } from 'app/plugins/da
 
 // if we have either "internal" or "both" configured this means the internal Alertmanager is receiving Grafana-managed alerts
 export const isInternalAlertmanagerInterestedInAlerts = (config?: GrafanaAlertingConfiguration): boolean => {
-  switch (config?.alertmanagersChoice) {
+  if (!config) {
+    // in case of no configuration, internal Alertmanager is interested in alerts
```

Comment:
nit: I think this one could use a better rationale, the comment currently describes what the code does but future me would be more interested in _why_ we assume the internal AM is interested when we have no config 

(because the backend doesn't have a configuration record in a new Grafana instance until the user has interacted with the config)

Reply:
good point